src/classes/PPlan_Primary_TDTM.cls

```diff
@@ -51,6 +61,16 @@ public with sharing class PPlan_Primary_TDTM extends TDTM_Runnable {
         Set<Id> primaryProgramIds = new Set<Id>();
         // BEFORE INSERT
         if ( newlist != null && triggerAction == TDTM_Runnable.Action.BeforeInsert && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.PPlan_Primary_TDTM_Before_Insert)) {
+            for (integer i=0; i<newlist.size(); i++) {
+                Program_Plan__c newPPlan = (Program_Plan__c)newlist[i];
+                if (newPPlan.Is_Primary__c == true) {
+                    // Make Plan "Current" if "Is Primary" is checked.
+                    newPPlan.Status__c = STATUS_CURRENT_VAL;
+                }
+            }
+        }
+        // AFTER INSERT
+        if ( newlist != null && triggerAction == TDTM_Runnable.Action.AfterInsert && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.PPlan_Primary_TDTM_After_Insert)) {
```

Comment:
The logic below was happening on BeforeInsert previously. Why are we changing it to AfterInsert?

Also, if it's now done in AfterInsert, wouldn't we need to add the records to DmlWrapper?